Dockerfile

```diff
@@ -1,15 +1,13 @@
-FROM python:3.9-slim
+FROM python:3.9
 
 RUN apt-get update \
-    && apt-get install -y sqlite3 \
+    && apt-get install -y \
+    sqlite3 \
+    rustc \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR /app
-
-COPY ./requirements.txt .
-
-RUN pip install -r ./requirements.txt
```

Comment:
I prefer this way round because it means only if you change the requirements it will run pip install, whereas now if you change any file (e.g. the python code) it will run pip install every time you rebuild